src/rules/noStaticOnlyClassesRule.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+import { isClassDeclaration, isFunctionWithBody } from "tsutils";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-static-only-classes",
+        description: Lint.Utils.dedent`
+            Disallows classes containing only static members. Classes
+            with non-empty constructors are ignored.`,
+        rationale: Lint.Utils.dedent`
+            Users who come from a Java-style OO language may wrap
+            their utility functions in an extra class, instead of
+            putting them at the top level.`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+
+    public static FAILURE_STRING = "Classes containing only static members are disallowed.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoStaticOnlyClassesWalker(sourceFile, this.ruleName, this.ruleArguments));
+    }
+}
+
+class NoStaticOnlyClassesWalker extends Lint.AbstractWalker<string[]> {
+    public walk(sourceFile: ts.SourceFile) {
+        for (const statement of sourceFile.statements) {
+            if (isClassDeclaration(statement)
+                && !hasExtendsClause(statement)
+                && !hasImplementsClause(statement)
+                && !isEmptyClass(statement)) {
+                for (const member of statement.members) {
+                    if (!hasStaticModifier(member.modifiers) && !isEmptyConstructor(member)) {
+                        return;
+                    }
+                }
+                if (statement.name !== undefined) {
```

Comment:
I didn't know this is a possibility. What should the failure message look like in this scenario? I've been underlining the class name to indicate a problem with the class itself.